### DISCUSSION TITLE
Encountered some problems with slow send transaction to zksync testnet chain

### DISCUSSION THREAD

**Discussion by easyhappy** [zkSync-Community-Hub/zksync-developers#26] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

the main question is that sometime take 20s ~ 40s+ to call some method like `eth_estimateGas` and `eth_sendRawTransaction` 

### What is the specific issue or error you're encountering?

1. Refer to the order in the picture below , it took 25s to send transaction to chain, and almost all of it was spent on calling eth_estimateGas. the transaction link is 

https://goerli.explorer.zksync.io/tx/0x89b95079ae0bd2d1ccbc28f3c2d914bcb776f039ca509894385f8ae7f23384bc

and sometime  some other transactions took 6~8s to call eth_sendRawTransaction

2. and we use `wss://testnet.era.zksync.dev/ws` and `https://testnet.era.zksync.dev/` in zksync testnet chain.  is the above slow problem caused by this rpc link?

3. sometimes, we call eth_getTransactionReceipt to get transaction receipt. it return the response as follow.

```
time="2023-06-09T04:06:00Z" level=info msg="HandleReceipt senderLog 15744, receipt status &{Type:0 PostState:[] Status:0 CumulativeGasUsed:0 Bloom:[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] Logs:[] TxHash:0x81679b15820427453e284b773640b75e9bb36c020256c56362107c60c45a2430 ContractAddress:0x0000000000000000000000000000000000000000 GasUsed:521865 BlockHash:0x0000000000000000000000000000000000000000000000000000000000000000 BlockNumber:<nil> TransactionIndex:0}"
time="2023-06-09T04:06:00Z" level=info msg="Create callback for log 15744 status FAILED, error: <nil>"
time="2023-06-09T04:06:00Z" level=info msg="log 0x81679b15820427453e284b773640b75e9bb36c020256c56362107c60c45a2430 receipt request finial failed"
```

About 1~2 seconds after this order was successfully uploaded to the chain, we found out that the receipt status of this response is still 0, BlockNumber is nil, so we mark this transaction to failed, but in the end we received order filled events, and at the same time about after another few tens of seconds, we manually inquired about this transaction and the receipt status became 1.

So is it understandable that if the receipt status is 0 and BlockNumber is nil, we can mark that the order is still in the initial state, not the failed state. 

![image](https://github.com/zkSync-Community-Hub/zkync-developers/assets/733348/5fea0c71-685a-44e9-ac5a-78bcb650b8e0)

### Can you share the error messages or logs you're receiving, if any?

No, just too slow

### Have you made any recent changes to the contract before encountering this issue?

No

### Are there any external libraries or contracts that your contract interacts with?

No

### Can you provide the relevant portions of your contract code where the issue is occurring?

No

### Have you tried to isolate the problem, and if so, what were the results?

No 

### What steps have you already taken to try to resolve the issue?

No

**Comment by idea404** [comment]
Hi @easyhappy can you confirm you're still experiencing this issue? If so, what RPC provider are you using?

**Reply by idea404** [reply]
@easyhappy I will close this issue in a week if not replied

**Comment by idea404** [comment, marked as answer]
Closing issue as left unreplied
